src/rust/cryptography-x509-validation/src/lib.rs

```diff
@@ -10,3 +10,14 @@ pub mod ops;
 pub mod policy;
 pub mod trust_store;
 pub mod types;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ValidationError {
+    Other(String),
```

Comment:
Also looks like `&'static str` should be sufficient here?

Reply:
Left a note about that here: https://github.com/pyca/cryptography/pull/9890#discussion_r1397981880

(I could do two variants, not sure how much better that is? Whatever your preference is 🙂)

Reply:
Let'd do `&'static str` for now, so this PR is just about changing the hierarchy, and we can do a `String` in future.